packages/e2e-test-utils/src/get-all-block-inserter-item-titles.js

```diff
@@ -9,6 +9,13 @@ import { sortBy, uniq } from 'lodash';
  * @return {Promise} Promise resolving with an array containing all inserter item titles.
  */
 export async function getAllBlockInserterItemTitles() {
+	// The inserter render lazy renders the list of blocks
+	// meaning we should wait for the browser to be completed idle.
+	// Ideally, we shouldn't use a timeout and instead check the browser is idle for
+	// a specific duration, but didn't manage to find a simple way to do that.
+	// eslint-disable-next-line no-restricted-syntax
+	await page.waitFor( 500 );
```

Comment:
> wait for all expected categories to appear instead of a wait.

This is exactly what I want to do here, but I didn't find a way to do it without introducing test specific code in the production code which I'm not a fan of.